src/Flysystem/AssetAdapter.php

```diff
@@ -195,4 +195,36 @@ protected function renderTemplate($template)
 
         return $result;
     }
+
```

Comment:
Maybe define those in an interface methods and fall back to the non-generator method when not implemented.

Reply:
What do you mean?